### ISSUE TITLE
[Feature]: Automated testing capabilities on Appsmith

### ISSUE THREAD

**Issue by danciaclara** [appsmithorg/appsmith#12358]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Today we manually test the app by running queries/functions in the respective pane/editor and checking the widget bindings, etc.. 

But it can become cumbersome to test everything manually and it's likely a few scenarios/test cases may be missed out. 

### Why should this be worked on?

Asked by a user on Discord

https://discord.com/channels/725602949748752515/725609493974614076/958354801844965486

**Comment by unknown1337** [comment]
+ one more request here, as apps grow the regression test becomes really cumbersome. Preferable I'd like a end-2-end functionality test, including e.g. interaction with UI elements (as I've already observed problems with the UI as well). Once this feature is going to be implemented, feel free to contact me for input in case desired;)

**Comment by LagunaElectric** [comment]
I don't think this should have been closed. The linked PR changes a link in the docs, and doesn't seem to have any connection to this issue other than being related to automated testing.
@rohan-arthur
